tools/generator/templates/player/rust/api/ffi.rs.jinja2

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2012-2020 Association Prologin <association@prologin.org>
+
+#![allow(clippy::unit_arg)]
+#![allow(clippy::unused_unit)]
+
+use crate::api;
+
+use std::ffi::{CStr, CString};
+use std::os::raw::{c_char, c_double, c_int, c_void};
+use std::{mem::{drop, size_of}, ptr, slice};
+
+#[allow(non_camel_case_types)]
+pub type c_bool = bool;
+
+/// Stechec2-specific array type.
+#[repr(C)]
+pub struct Array<T> {
+    ptr: *mut T,
+    len: usize,
+}
+
+impl<T> Drop for Array<T> {
+    fn drop(&mut self) {
+        unsafe {
+            slice::from_raw_parts_mut(self.ptr, self.len)
+                .iter_mut()
+                .for_each(drop);
+            free(self.ptr as _);
+        }
+    }
+}
+
+/// Represents an owned C string that was created by a foreign function using
+/// malloc. This means that this string must be deallocated using free.
+#[repr(C)]
+pub struct RawString {
+    ptr: *mut c_char
+}
+
+impl Drop for RawString {
+    fn drop(&mut self) {
+        unsafe {
+            free(self.ptr as _);
+        }
+    }
+}
+
+// Enums
+
+{% for enum in game.enum %}
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub enum {{ enum.enum_name|camel_case }} {
+    {% for field_name, _ in enum.enum_field %}
+    {{ field_name|camel_case }},
+    {% endfor %}
+}
+
+{% endfor -%}
+
+// Structures
+
+{% for struct in game.struct %}
+#[repr(C)]
+pub struct {{ struct.str_name|camel_case }} {
+    {% for field_name, field_type, _ in struct.str_field %}
+    {{ field_name }}: {{ field_type|rust_ffi_type }},
+    {% endfor %}
+}
+
+{% endfor -%}
+
+// Conversion traits
+
+pub trait ToAPI<T> {
```

Comment:
Do you think it could be a good idea to reuse the nomenclature used for the dynamic language interfaces here? Like CToRust and RustToC

Reply:
If I can help consistency I don't have bigger cons than "it takes a few more letters to write".

However I think that it is still more idiomatic to write `var.to_rust()` than `var.rust_to_c()`, maybe inner methods should keep this pattern.

Reply:
I understand but right now it's pretty hard to keep track of what is of what type, and maybe the explicit rust_to_c() could help